rules/opentitan.bzl

```diff
@@ -166,14 +181,120 @@ def _elf_to_scrambled(ctx):
         )
         return [DefaultInfo(files = depset(outputs), data_runfiles = ctx.runfiles(files = outputs))]
 
-elf_to_scrambled = rule(
-    implementation = _elf_to_scrambled,
+elf_to_scrambled_rom_vmem = rule(
+    implementation = _elf_to_scrambled_rom_impl,
     cfg = opentitan_transition,
     attrs = {
         "srcs": attr.label_list(allow_files = True),
         "platform": attr.string(default = OPENTITAN_PLATFORM),
-        "_tool": attr.label(default = "//hw/ip/rom_ctrl/util:scramble_image.py", allow_files = True),
-        "_config": attr.label(default = "//hw/top_earlgrey/data:autogen/top_earlgrey.gen.hjson", allow_files = True),
+        "_tool": attr.label(
+            default = "//hw/ip/rom_ctrl/util:scramble_image.py",
+            allow_files = True,
+        ),
+        "_config": attr.label(
+            default = "//hw/top_earlgrey/data:autogen/top_earlgrey.gen.hjson",
+            allow_files = True,
+        ),
+        "_allowlist_function_transition": attr.label(
+            default = "@bazel_tools//tools/allowlists/function_transition_allowlist",
+        ),
+    },
+)
+
+def _bin_to_flash_vmem_impl(ctx):
+    outputs = []
+    vmem = ctx.actions.declare_file("{}.{}.vmem".format(
+        # Remove ".bin" from file basename.
+        ctx.file.bin.basename[:-4],
+        ctx.attr.word_size,
+    ))
+    outputs.append(vmem)
+    ctx.actions.run(
+        outputs = [vmem],
+        inputs = [
+            ctx.file.bin,
+        ],
+        arguments = [
+            ctx.file.bin.path,
+            "--binary",
+            # Reverse the endianness of every word.
+            "--offset",
+            "0x0",
+            "--byte-swap",
+            str(ctx.attr.word_size // 8),
+            # Pad to word alignment.
+            "--fill",
+            "0xff",
+            "-within",
+            ctx.file.bin.path,
+            "-binary",
+            "-range-pad",
+            str(ctx.attr.word_size // 8),
+            # Output a VMEM file with specified word size
+            "--output",
+            vmem.path,
+            "--vmem",
+            str(ctx.attr.word_size),
+        ],
+        executable = "srec_cat",
```

Comment:
Comment: this executable is expected to be installed (as required by the srecord package in apt-requirements.txt)

Reply:
Fixed.